### PR TITLE
Migrate to DI

### LINKED ISSUE #42: Migrate fusion-plugin-i18n-react to createPlugin

### PR DESCRIPTION
**Pull request by ganemone**
Fixes #42

### CODE REVIEW

File: src/__tests__/index.node.js [hunk 1]
```diff
@@ -1,25 +1,56 @@
...
-  const data = {test: 'hello', interpolated: 'hi ${value}'};
-
-  t.equals(typeof Plugin, 'function');
-
-  const I18n = Plugin({
-    loadTranslations: () => ({translations: data, locale: 'en-US'}),
-  });
-  const ctx = {headers: {'accept-language': 'en-US'}, memoized: new Map()};
-  t.equals(typeof I18n.of(ctx).translate, 'function');
+test('exports', t => {
```
Comment (AlexMSmithCA) [inline src/__tests__/index.node.js]: Cool!  Do we want to do this with all plugins?  We haven't been consistent so far.

Comment (ganemone) [reply src/__tests__/index.node.js]: I kinda like having this, but idk

Comment (AlexMSmithCA) [reply src/__tests__/index.node.js]: I think it's a nice to have.   Useful for adding a minimal amount of friction to removing an export and making that major breaking change.

**Comment by ganemone** [comment]
!merge

**Pull request merged**